src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/NoopTelemetryModule.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Microsoft.Azure.WebJobs.Logging.ApplicationInsights
+{
+    /// <summary>
+    /// Noop telemetry module that is added instead of another one, which is disabled by settings.
+    /// </summary>
+    /// <typeparam name="T">Type of the module substituted by this Noop module. Noop module is added to DI as singleton and generic type makes it unique.</typeparam>
+    class NoopTelemetryModule<T> : ITelemetryModule
```

Comment:
> 2. If you use a singleton Instance (like NullLogger.Instance), can you remove the generic?

The problem is that we are going to use it like:

```csharp
services.AddSingleton<ITelemetryModule>(provider => {
  //....
  return new NullTelemetryModule<RequestTrackingTelemetryModule>();
});

services.AddSingleton<ITelemetryModule>(provider => {
  //....
  return new NullTelemetryModule<PerformanceCounters>();
});

services.AddSingleton<ITelemetryModule>(provider => {
  //....
  return new NullTelemetryModule<Dependencies>();
});

```

If we use NullTelemetryModule without generic, we'll get the exception because we are adding more than one singleton of the same service and implementation type.

If we use TryAddSingleton<ITelemetryModule>(factory), we'll not be able to add more than one ITelemetryModule of any type.

So generics is a feature :)
The way to avoid them is to return null and expect null in GetServices<ITelemetryModule>. I wanted to avoid it since users also have access to DI and we might create some confusion here.

Reply:
What exception do you get? I just tested this with the .Instance approach and it looks like it worked:
```
services.AddSingleton<ITelemetryModule>(provider =>
{
    return NoopTelemetryModule.Instance;
});

services.AddSingleton<ITelemetryModule>(provider =>
{
    return NoopTelemetryModule.Instance;
});
```

And then when I run, I just see two (of the same) NoopTelemetryModules in the collection:
![image](https://user-images.githubusercontent.com/1089915/54050074-29469400-4193-11e9-952e-73f8a13118a0.png)

I thought we did this elsewhere with `NullLoggerProvider.Instance`, which is why I suggested it. But now I can't find it, so maybe I'm misremembering.

Reply:
sorry, perhaps I was testing something else that threw on adding singleton. Changed this to the NullTelemetryModule.Instance approach and it worked. Thanks a lot